crates/router/src/core/payouts/helpers.rs

```diff
@@ -194,6 +194,50 @@ pub async fn make_payout_method_data<'a>(
     }
 }
 
+pub fn should_create_connector_transfer_method(
+    payout_data: &PayoutData,
+    connector_data: &api::ConnectorData,
+) -> RouterResult<Option<String>> {
+    let connector_mandate_id = if let Some(pm) = &payout_data.payment_method {
+        #[cfg(all(
+            any(feature = "v1", feature = "v2"),
+            not(feature = "payment_methods_v2")
+        ))]
+        let connector_mandate_details = pm
+            .connector_mandate_details
+            .clone()
+            .map(|details| {
+                details.parse_value::<diesel_models::PaymentsMandateReference>(
```

Comment:
Here you are returning a struct `PaymentMandateReference` for V1, whereas in V2 you are returning the field `connector_mandate_id` in the struct `PaymentMandateReference`, while this function has a return type as a String